examples/input-handling-multiple.rs

```diff
@@ -0,0 +1,125 @@
+        .draw(display)?;
+
+    Ok(())
+}
+
+enum LoopResult {
+    Continue,
+    AddNewWindow,
+    Quit,
+}
+
+fn handle_loop(
+    window: &mut Window,
+    display: &mut SimulatorDisplay<Rgb888>,
+    position: &mut Point,
+) -> Result<LoopResult, core::convert::Infallible> {
+    window.update(display);
+
+    let events = window.events();
+    for event in events {
+        match event {
+            SimulatorEvent::Quit => return Ok(LoopResult::Quit),
+            SimulatorEvent::KeyDown { keycode, .. } => {
+                let delta = match keycode {
+                    Keycode::Left => Point::new(-KEYBOARD_DELTA, 0),
+                    Keycode::Right => Point::new(KEYBOARD_DELTA, 0),
+                    Keycode::Up => Point::new(0, -KEYBOARD_DELTA),
+                    Keycode::Down => Point::new(0, KEYBOARD_DELTA),
+                    Keycode::N => return Ok(LoopResult::AddNewWindow),
+                    _ => Point::zero(),
+                };
+                let new_position = *position + delta;
+                move_circle(display, *position, new_position)?;
+                *position = new_position;
+            }
+            SimulatorEvent::MouseButtonUp { point, .. } => {
+                move_circle(display, *position, point)?;
+                *position = point;
+            }
+            _ => {}
+        }
+    }
+    Ok(LoopResult::Continue)
+}
+
+fn main() -> Result<(), core::convert::Infallible> {
+    let mut windows = vec![(
+        SimulatorDisplay::new(Size::new(800, 480)),
+        Window::new(
+            "Click to move circle (press N for new window)",
+            &OutputSettings::default(),
+        ),
+        Point::new(200, 200),
+    )];
+
+    for (display, _, position) in windows.iter_mut() {
+        Circle::with_center(*position, 200)
+            .into_styled(PrimitiveStyle::with_fill(FOREGROUND_COLOR))
+            .draw(display)?;
+    }
+
+    'running: loop {
+        for (display, window, position) in windows.iter_mut() {
+            match handle_loop(window, display, position)? {
+                LoopResult::Continue => {}
+                LoopResult::AddNewWindow => {
+                    let mut display = SimulatorDisplay::new(Size::new(800, 480));
+                    let window = Window::new(
+                        "Click to move circle (press N for new window)",
+                        &OutputSettings::default(),
+                    );
+                    let position = Point::new(200, 200);
+
+                    Circle::with_center(position, 200)
+                        .into_styled(PrimitiveStyle::with_fill(FOREGROUND_COLOR))
+                        .draw(&mut display)?;
+
+                    windows.push((display, window, position));
+                    break;
+                }
+                LoopResult::Quit => {
+                    break 'running;
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
```

Comment:
I tried to remove the duplicated code to create the window and changed the comments a bit. I would also suggest that we rename this file to `multiple-windows.rs`, because that's the primary functionality it demonstrates. The original name contained `multiple`, but in my opinion it wasn't clear that this refers to windows and not a special type of input handling.
```suggestion
//! # Example: Multiple simulator windows
//!
//! This example demonstrates how multiple simulator windows can be opened at
//! the same time and how each window has it's own event queue. Press `n` to
//! open an additional window. The circle in each window can be moved independently
//! with the arrow keys or by clicking with the mouse.

use embedded_graphics::{
    pixelcolor::Rgb888,
    prelude::*,
    primitives::{Circle, PrimitiveStyle},
};
use embedded_graphics_simulator::{
    sdl2::Keycode, OutputSettings, SimulatorDisplay, SimulatorEvent, Window,
};

const BACKGROUND_COLOR: Rgb888 = Rgb888::BLACK;
const FOREGROUND_COLOR: Rgb888 = Rgb888::RED;
const KEYBOARD_DELTA: i32 = 20;
const CIRCLE_DIAMETER: u32 = 200;

enum LoopResult {
    Continue,
    AddNewWindow,
    Quit,
}

struct ExampleWindow {
    display: SimulatorDisplay<Rgb888>,
    window: Window,
    circle_position: Point,
}

impl ExampleWindow {
    fn new() -> Self {
        Self {
            display: SimulatorDisplay::new(Size::new(800, 480)),
            window: Window::new(
                "Click to move circle (press N for new window)",
                &OutputSettings::default(),
            ),
            circle_position: Point::zero(),
        }
    }

    fn init(&mut self) -> Result<(), core::convert::Infallible> {
        self.circle_position = Point::new(200, 200);

        // Draw the initial circle.
        Circle::with_center(self.circle_position, CIRCLE_DIAMETER)
            .into_styled(PrimitiveStyle::with_fill(FOREGROUND_COLOR))
            .draw(&mut self.display)?;

        Ok(())
    }

    fn handle_events(&mut self) -> Result<LoopResult, core::convert::Infallible> {
        self.window.update(&self.display);

        let events = self.window.events();
        for event in events {
            let new_position = match event {
                SimulatorEvent::Quit => return Ok(LoopResult::Quit),
                SimulatorEvent::KeyDown { keycode, .. } => {
                    let delta = match keycode {
                        Keycode::Left => Point::new(-KEYBOARD_DELTA, 0),
                        Keycode::Right => Point::new(KEYBOARD_DELTA, 0),
                        Keycode::Up => Point::new(0, -KEYBOARD_DELTA),
                        Keycode::Down => Point::new(0, KEYBOARD_DELTA),
                        Keycode::N => return Ok(LoopResult::AddNewWindow),
                        _ => Point::zero(),
                    };

                    Some(self.circle_position + delta)
                }
                SimulatorEvent::MouseButtonUp { point, .. } => Some(point),
                _ => None,
            };

            if let Some(new_position) = new_position {
                // Clear old circle.
                Circle::with_center(self.circle_position, CIRCLE_DIAMETER)
                    .into_styled(PrimitiveStyle::with_fill(BACKGROUND_COLOR))
                    .draw(&mut self.display)?;

                self.circle_position = new_position;

                // Draw circle at new location.
                Circle::with_center(self.circle_position, CIRCLE_DIAMETER)
                    .into_styled(PrimitiveStyle::with_fill(FOREGROUND_COLOR))
                    .draw(&mut self.display)?;
            }
        }
        Ok(LoopResult::Continue)
    }
}

fn main() -> Result<(), core::convert::Infallible> {
    let mut first_window = ExampleWindow::new();
    first_window.init()?;

    let mut windows = vec![first_window];

    'running: loop {
        for window in windows.iter_mut() {
            match window.handle_events()? {
                LoopResult::Continue => {}
                LoopResult::AddNewWindow => {
                    let mut window = ExampleWindow::new();
                    window.init()?;

                    // The `windows` array cannot be modified while it is
                    // mutably borrowed by the `iter_mut` iterator. By breaking
                    // the loop over the windows the lifetime of the iterator is
                    // limited and does no longer interfere with the `push`.
                    windows.push(window);
                    break;
                }
                LoopResult::Quit => {
                    break 'running;
                }
            }
        }
    }

    Ok(())
}
```

Reply:
Done, along with a PR to fix the close a window problem.